File: prep/c/exercism/armstron_number/armstrong.c
#include <stdio.h>
#include <math.h>

int main()
{
  int orig, n, digit;
  int count, running_total; 
  running_total = count = 0;
  printf("Enter a number: ");
  scanf("%d", &n);
  orig = n;

  /* get length of input */ 
  while(n != 0)
  {
    n = n/10;
    count++;
  }

  n = orig 

  /* calculate running_total */ 
  while(n != 0)
  {
    digit = n % 10; 
    digit = pow(digit, count);
    running_total += digit;
    n = n / 10;
  }

  return running_total == orig;
  

}

File: prep/c/exercism/armstron_number/armstrong.py
def armstrong(number):
    running_total = 0
    digit_list = []
    orig = number

    while number > 0:
        digit_list.append(number % 10)
        number = number // 10

    power = len(digit_list)
    for digit in digit_list:
        exponent = digit**power
        running_total += exponent

    return running_total == orig


print(armstrong(10))
